cyral/internal/repository/resource_cyral_repository.go

```diff
@@ -263,10 +279,18 @@ func (r *RepoInfo) MongoDBSettingsFromInterface(i []interface{}) error {
 			Standalone,
 		)
 	}
+	if serverType == Sharded && mongoFlavor == MongoDBFlavorDocumentDB {
```

Comment:
We already have validation for this combination in `crud`, so I don't think we should do it here too. I feel like the terraform provider should not encode these validations, but rather they should be left to the gRPC/REST implementation.

That said, that's just my opinion, and I don't know what's the idiomatic thing to do. Let's see what @wcmjunior thinks.

Reply:
I am fine with both, we can wait for wilson.

Reply:
It is ok to have the validation only in the backend as long as it provides an error message that actually helps the user to solve the problem. If it returns something like `Invalid parameter` it does not help, but if it returns something in the lines of `Parameter X must be 'Y' or 'Z'` it is ok.